src/openagi/agent.py

```diff
@@ -112,7 +127,7 @@ def execute(
             all_tasks=all_tasks,
             current_task_name=task.name,
             current_description=task.description,
-            previous_task=None,  # self.st_memory.get_previous_task()
+            previous_task=self.lt_memory.search(task.name),
```

Comment:
1. use `task.id`
2. get previous task data from  `st_memory` of the curr_objective.